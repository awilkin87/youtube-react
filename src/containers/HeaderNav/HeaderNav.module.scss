// Global specificity to override Semantic UI
:global(.ui.menu).topMenu {
  border: none; // Remove Semantic UI bottom shadow border
  padding: 8px 15px;

  div {
    padding: 0;
  }

  .left img {
    width: 90px;
  }

  .center {
    flex-grow: 1;
    justify-content: center;
    margin: 0px 10px;

    .search {
      flex-grow: 0.5;
    }
  }

  // Class added by Semantic UI 
  :global(.right) {
    gap: 10px;

    // Class added by Semantic UI 
    :global(.item) {
      min-width: 32px;
      min-height: 32px;
      align-content: center;
      justify-content: center;
    }

    .headerIcon {
      color: #a0a0a0;
      margin: 0px
    }

    // Class added by Semantic UI 
    :global(.avatar) {
      margin: 0px;
    }
  }
}