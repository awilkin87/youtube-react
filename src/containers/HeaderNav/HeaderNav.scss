.ui.top-menu.menu {
  border: none; /*Remove Semantic UI's bottom shadow border */
  padding: 8px 15px;

  div {
    padding: 0;
  }

  .left img {
    width: 90px;
  }

  .center {
    flex-grow: 1;
    justify-content: center;
    margin: 0px 10px;

    .search {
      flex-grow: 0.5;
    }
  }

  .right {
    gap: 10px;

    .item {
      min-width: 32px;
      min-height: 32px;
      align-content: center;
      justify-content: center;
    }

    .header-icon {
      color: #a0a0a0;
      margin: 0px
    }

    .avatar {
      margin: 0px;
    }
  }
}