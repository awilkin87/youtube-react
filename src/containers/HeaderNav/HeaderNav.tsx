import React from 'react';
import { Link } from 'react-router-dom';
import { Form, Image, Input, Menu, Icon } from 'semantic-ui-react';
import logo from '../../assets/images/logo.jpg';
import styles from './HeaderNav.module.scss';

/** Top nav bar. */
export default function HeaderNav() {
  return (
    <Menu borderless className={styles.topMenu} fixed='top'>
      <Menu.Item header className={styles.left}>
        <Link to='/'>
          <Image src={logo} size='tiny' />
        </Link>
      </Menu.Item>

      <Menu.Item className={styles.center}>
        <Form className={styles.search}>
          <Form.Field>
            <Input placeholder='Search' size='small' action='Go' />
          </Form.Field>
        </Form>
      </Menu.Item>

      <Menu.Item position='right'>
        <Menu.Item>
          <Icon className={styles.headerIcon} name='video camera' size='large' />
        </Menu.Item>
        <Menu.Item>
          <Icon className={styles.headerIcon} name='grid layout' size='large' />
        </Menu.Item>
        <Menu.Item>
          <Icon className={styles.headerIcon} name='chat' size='large' />
        </Menu.Item>
        <Menu.Item>
          <Icon className={styles.headerIcon} name='alarm' size='large' />
        </Menu.Item>
        <Menu.Item name='avatar'>
          <Image src='http://via.placeholder.com/80x80' avatar />
        </Menu.Item>
      </Menu.Item>
    </Menu>
  );
}