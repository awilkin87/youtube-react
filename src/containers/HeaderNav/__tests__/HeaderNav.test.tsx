import { shallow } from "enzyme";
import HeaderNav from '../HeaderNav';

test('Renders HeaderNav', () => {
  const wrapper = shallow(<HeaderNav />);
  expect(wrapper).toMatchSnapshot();
});