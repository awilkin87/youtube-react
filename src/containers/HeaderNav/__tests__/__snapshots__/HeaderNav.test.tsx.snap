// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders HeaderNav 1`] = `
<Menu
  borderless={true}
  className="topMenu"
  fixed="top"
>
  <MenuItem
    className="left"
    header={true}
  >
    <Link
      to="/"
    >
      <Image
        as="img"
        size="tiny"
        src="logo.jpg"
        ui={true}
      />
    </Link>
  </MenuItem>
  <MenuItem
    className="center"
  >
    <Form
      as="form"
      className="search"
    >
      <FormField>
        <Input
          action="Go"
          placeholder="Search"
          size="small"
          type="text"
        />
      </FormField>
    </Form>
  </MenuItem>
  <MenuItem
    position="right"
  >
    <MenuItem>
      <Icon
        as="i"
        className="headerIcon"
        name="video camera"
        size="large"
      />
    </MenuItem>
    <MenuItem>
      <Icon
        as="i"
        className="headerIcon"
        name="grid layout"
        size="large"
      />
    </MenuItem>
    <MenuItem>
      <Icon
        as="i"
        className="headerIcon"
        name="chat"
        size="large"
      />
    </MenuItem>
    <MenuItem>
      <Icon
        as="i"
        className="headerIcon"
        name="alarm"
        size="large"
      />
    </MenuItem>
    <MenuItem
      name="avatar"
    >
      <Image
        as="img"
        avatar={true}
        src="http://via.placeholder.com/80x80"
        ui={true}
      />
    </MenuItem>
  </MenuItem>
</Menu>
`;
